crates/engine/tree/src/persistence.rs

```diff
@@ -192,15 +192,16 @@ pub enum PersistenceAction {
 
 /// A handle to the persistence service
 #[derive(Debug, Clone)]
-pub struct PersistenceHandle {
+pub struct PersistenceHandle<N: NodePrimitives = EthPrimitives> {
     /// The channel used to communicate with the persistence service
     sender: Sender<PersistenceAction>,
+    _marker: std::marker::PhantomData<N>,
```

Comment:
in a next step, we need to apply the same to PersistenceAction, then we can get rid of the phantomdata